bot/admin/test/xray/src/main/kotlin/XrayService.kt

```diff
@@ -582,15 +584,75 @@ class XrayService(
             testsPlans: List<String>,
             labelTestPlansMap: Map<String, String> = emptyMap()
     ): XrayTest? {
+        val steps = defineTestSteps(dialog)
+
+        val labels = linkedJira?.let { XrayClient.getLabels(it) } ?: emptyList()
+
+        val test = JiraTest(
+                jiraProject,
+                testName.invoke(labels),
+                "",
+                listOf("automatisation"),
+                testTypeField,
+                manualStepsField,
+                "cucumber-tock"
+        )
+        val jira = XrayClient.createTest(test)
+        if (linkedJira != null) {
+            XrayClient.linkTest(jira.key, linkedJira)
+            testsPlans.forEach {
+                XrayClient.addTestToTestPlan(jira.key, it)
+            }
+            if (labelTestPlansMap.isNotEmpty()) {
+                labels
+                        .filter { labelTestPlansMap.containsKey(it) }
+                        .forEach {
+                            XrayClient.addTestToTestPlan(jira.key, labelTestPlansMap[it]!!)
+                        }
+            }
+        }
+
+        steps?.forEach {
+            XrayClient.saveStep(jira.key, it)
+        }
+
+        XrayPrecondition.getPreconditionForUserInterface(dialog.actions.first().userInterfaceType)
+                ?.apply {
+                    XrayClient.addPrecondition(this, jira.key)
+                }
+
+        return XrayTest(jira.key)
+    }
+
+    fun updateXrayTest(dialog: DialogReport, testKey: String): XrayTest? {
+        val steps = defineTestSteps(dialog)
+
+        if (XrayClient.getIssue(testKey) == null) {
+            return null
```

Comment:
it seems this return is unreachable because `getIssue` returns `error` if we are unable to get the issue

Reply:
I've removed the nullity test but left the getIssue() call to be able to make the difference between "issue does not exist" or "no steps available for the current test" errors.